### PR TITLE
WW5052 TypeConversionException instead StrutsException

### PR DESCRIPTION
**Pull request by santosr09**
The StrutsException was switched to TypeConversionException in these classes:
 - DateConverter
 - NumberConverter
 - XWorkBasicConverter

Required tests were added in these classes:
 - DateConverterTest (New class)
 - NumberConverterTest
 - XWorkBasicConverterTest

### CODE REVIEW

File: core/src/test/java/com/opensymphony/xwork2/conversion/impl/DateConverterTest.java [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	
+	private final static String TIME_00_59_10 = "00:59:10";
+	private final static String TIMESTAMP_STR = "2020-03-20 00:59:10";
+	private final static String DATE_STR = "2020-03-20";
+	private final static String DATE_CONVERTED = "Fri Mar 20 00:00:00 CST 2020";
+	private final static String INVALID_DATE = "99/99/2010";
+	private final static String MESSAGE_PARSE_ERROR = "Could not parse date";
+	private final static String MESSAGE_DEFAULT_CONSTRUCTOR_ERROR = "Couldn't create class null using default (long) constructor";
+	
+	public void  testSqlTimeType(){
```
Comment (sepe81) [inline core/src/test/java/com/opensymphony/xwork2/conversion/impl/DateConverterTest.java]:
```suggestion
	public void testSqlTimeType() {
```

Comment (sepe81) [reply core/src/test/java/com/opensymphony/xwork2/conversion/impl/DateConverterTest.java]: Some more similar lines need formatting.

**Comment by lukaszlenart** [comment]
Nice, superb :)

Just one note, would you mind moving `TypeConversionException` into `org.apache.struts2.conversion` package?

**Comment by santosr09** [comment]
Done! Thanks for your comments and support, Team
Regards

**Pull request merged**